### ISSUE TITLE
BuildOptimizer `0.6.0` not exporting enum

### ISSUE THREAD

**Issue by boltzj** [angular/devkit#859]
### Bug Report or Feature Request 
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Area
```
- [X] devkit
- [ ] schematics
```

### Versions

Node version
```
Node : v8.11.1
```
Cli version:
```
@angular-devkit/architect         0.6.0
@angular-devkit/build-angular     0.6.0
@angular-devkit/build-optimizer   0.6.0
@angular-devkit/core              0.6.0
@angular-devkit/schematics        0.6.0
@angular/cdk                      5.2.5
@angular/material                 5.2.5
@schematics/angular               0.5.13
@schematics/package-update        0.5.13
@schematics/update                0.6.0
rxjs                              6.1.0
typescript                        2.7.2
webpack                           4.7.0
```

### Repro steps

[Here is the repo](https://github.com/boltzj/angularv6-uirouter) to reproduce the issue.

#### Steps to reproduce the issue
  - New project from CLI (`6.0.0`)
  - Add `rxjs-compat`
  - Add the 3rd party router and bootstrap the UIView in the app module

The project work correctly until you activate the build with AOT & buildOptimizer

#### Version of build-optimizer affected
It's working correctly if you force `@angular-devkit/build-optimizer` in version `0.5.3` instead of `0.6.0`

### The log given by the failure

```
Uncaught TypeError: Cannot read property 'STATE' of undefined
```

After importing a module exporting those enums:
```
export enum TransitionHookPhase {
  CREATE,
  BEFORE,
  RUN,
  SUCCESS,
  ERROR,
}
export enum TransitionHookScope {
  TRANSITION,
  STATE,
}
```
```
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TransitionHookPhase;
(function (TransitionHookPhase) {
    TransitionHookPhase[TransitionHookPhase["CREATE"] = 0] = "CREATE";
    TransitionHookPhase[TransitionHookPhase["BEFORE"] = 1] = "BEFORE";
    TransitionHookPhase[TransitionHookPhase["RUN"] = 2] = "RUN";
    TransitionHookPhase[TransitionHookPhase["SUCCESS"] = 3] = "SUCCESS";
    TransitionHookPhase[TransitionHookPhase["ERROR"] = 4] = "ERROR";
})(TransitionHookPhase = exports.TransitionHookPhase || (exports.TransitionHookPhase = {}));
var TransitionHookScope;
(function (TransitionHookScope) {
    TransitionHookScope[TransitionHookScope["TRANSITION"] = 0] = "TRANSITION";
    TransitionHookScope[TransitionHookScope["STATE"] = 1] = "STATE";
})(TransitionHookScope = exports.TransitionHookScope || (exports.TransitionHookScope = {}));
//# sourceMappingURL=interface.js.map%
```
with build optimization the module's `exports` just contain the `__esModule : true` property

### Desired functionality
Export of the enum has expected

### Mention any other details that might be useful
Here is the original issue on [ui-router/core](https://github.com/ui-router/core/issues/171)
Here is the [reproduction repo](https://github.com/boltzj/angularv6-uirouter)

**Comment by ramandeep-singh-1983** [comment]
I have the same issue, i am using ng2-logger in my application. First, i saw error in that, then i updated tsconfig.app.json to include the ng2-logger ts files:

"include": [
    "../node_modules/angular2-logger/app/core/level.ts",
    "../node_modules/angular2-logger/app/core/logger.ts",
    "../node_modules/angular2-logger/app/core/providers.ts"
]

This fixed the logger error but then i got another error in msalx library. Putting buildOptimizer to false fixes this.

If this is an issue, what is the best workaround:

Adding ts code references in tsconfig.app.json
OR
Disabling buildOptimizer
OR
Using buildOptimizer version 0.5.3

**Comment by clydin** [comment]
Resolved via #878
